core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGeneData.java

```diff
@@ -72,119 +72,190 @@ public static void importData(File geneFile, String genomeBuild) throws IOExcept
+                    gene = new CanonicalGene(entrezGeneId, geneSymbol, aliases);
+                    refGene = new ReferenceGenomeGene(entrezGeneId, referenceGenomeId, chr, cytoband, 0, 0, 0);
+                    Set<CanonicalGene> genes = genesWithoutSymbolFromNomenClatureAuthority.get(geneSymbol);
+                    Set<ReferenceGenomeGene> refGenes = refGenesWithoutSymbolFromNomenClatureAuthority.get(geneSymbol);
+                    if (genes==null) {
+                        genes = new HashSet<CanonicalGene>();
+                        genesWithoutSymbolFromNomenClatureAuthority.put(geneSymbol, genes);
                     }
+                    if (refGenes==null) {
+                        refGenes = new HashSet<ReferenceGenomeGene>();
+                        refGenesWithoutSymbolFromNomenClatureAuthority.put(geneSymbol, refGenes);
+                    }
+                    genes.add(gene);
+                    refGenes.add(refGene);
+                }
+
+                if (gene!=null) {
+                    gene.setType(type);
+                }
             }
             addGenesToDB(genesWithSymbolFromNomenClatureAuthority, genesWithoutSymbolFromNomenClatureAuthority);
-            addReferenceGenomeGenesToDB(refGenesWithSymbolFromNomenClatureAuthority, 
+            addReferenceGenomeGenesToDB(refGenesWithSymbolFromNomenClatureAuthority,
                 refGenesWithoutSymbolFromNomenClatureAuthority);
         }
     }
-        
-        /**
-         * Iterate over the genes found in the given maps and try to add them to the DB.
-         * 
-         * @param genesWithSymbolFromNomenClatureAuthority: genes with official symbol
-         * @param genesWithoutSymbolFromNomenClatureAuthority: genes without official symbol (can happen, some entrez genes
-         * have no official symbol yet, but a temporary symbol).
-         * 
-         * @throws DaoException
-         */
-        private static void addGenesToDB(Map<String, Set<CanonicalGene>> genesWithSymbolFromNomenClatureAuthority,
-                  Map<String, Set<CanonicalGene>> genesWithoutSymbolFromNomenClatureAuthority) throws DaoException {
-                     
-          DaoGeneOptimized daoGene = DaoGeneOptimized.getInstance();
-          int nrExisting = 0;
+
+    /**
+     * Adds the genes parsed from the HGNC file into the database.
+     *
+     * @param geneFile HGNC File with gene information
+     * @throws IOException
+     * @throws DaoException
+     */
+    public static void importHGNCData(File geneFile, String genomeBuild) throws IOException, DaoException {
+        Map<String, Set<CanonicalGene>> canonicalGeneMap = new LinkedHashMap<>();
+        Map<String, Set<ReferenceGenomeGene>> refGeneMap = new LinkedHashMap<>();
+        try (FileReader reader = new FileReader(geneFile)) {
+            BufferedReader buf = new BufferedReader(reader);
+            String line;
+            buf.readLine(); // skip first (header) line
```

Comment:
because we do not parse the line to distinguish a header / comment / blank line, the documentation should be emphatic that the input file must have a single line header, with no extra blank lines or comments

Reply:
@yichaoS : we are not sure where the file format for the hgnc gene file is documented ... maybe you can check if it says somewhere that there *must* be a header line and that it should be only a single line

Reply:
@sheridancbio @mandawilson Yes, the hgnc gene file (gene_info.txt) needs to have a header and it's a single line. Here is where it resides and looks like: https://raw.githubusercontent.com/cBioPortal/datahub-study-curation-tools/master/gene-table-update/build-input-for-importer/gene_info.txt